test/authentication/authentication.js

```diff
@@ -39,6 +39,24 @@ describe("mCAP.authentication", function () {
     expect(user.hasRoleWithUuid('GROUP_NAME_ONE')).toBe(true);
     expect(user.hasRoleWithUuid('GROUP_NAME_TWO')).toBe(false);
   });
+
+  it('ignores custom attributes with empty strings', function() {
+    var user = new AuthenticatedUser({uuid: '09F40863-DD69-49AE-9F8C-C83EA3BCD6BA'});
+    user.set('preferences', new UserPreferences());
+
+    user.setReferencedCollections({
```

Comment:
actually you would call
````
user.get('preferences').set({
  custom1: '',
  ...
});
```
Set referencedCollection should not be called. The method should only be called by `parse`and `set` to make sure that the nested attributes are not overwritten